kvm.yaml

```
@@ -217,6 +217,57 @@ spec:
 ---
 apiVersion: release.giantswarm.io/v1alpha1
 kind: Release
+metadata:
+  annotations:
+    giantswarm.io/docs: https://docs.giantswarm.io/reference/cp-k8s-api/releases.release.giantswarm.io/
+  name: v9.0.1
+spec:
+  state: active
+  date: 2020-04-24T12:00:00Z
+  apps:
+  - name: cert-exporter
+    version: 1.2.2
+  - name: chart-operator
+    version: 0.13.0
+  - name: coredns
+    componentVersion: 1.6.5
+    version: 1.1.8
+  - name: kube-state-metrics
+    componentVersion: 1.9.2
+    version: 1.0.4
```

```suggestion
    componentVersion: 1.9.5
    version: 1.0.5
```
Are you sure KSM 1.0.5/1.9.5 is not compatible with k8s 1.15? Seems to work well for equivalent aws release https://github.com/giantswarm/releases/blob/master/aws.yaml#L711-L761

If suggestion is accepted, release notes for 1.0.5 have to be added too.

---

According to their compatibility matrix, 1.9.5 is k8s 1.16. I was running it accidentally with this release for a while without noticeable issues, but unless there's a big benefit for us I'd keep it at the supported version. This is already a lot of change

---

There is big benefit - 1.9.5 works while 1.9.2 is crashing and paging https://github.com/giantswarm/giantswarm/issues/9506

It's compatible with 1.16 and it works well with 1.15 too.

---

Alright, if we know it works I'll go ahead and add it back

---

For us when testing it did not work on kvm 1.15 - we can try again but it was full of error logs when we tried.

---

Yes @sslavic looks like there is one of these per second in the ksm logs:
```
E0424 07:47:30.318475       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191109102209-3c0d1af94be5/tools/cache/reflector.go:108: Failed to list *v1.ValidatingWebhookConfiguration: the server could not find the requested resource
```

---

It's definitely a change on the upstream side 
https://github.com/kubernetes/kube-state-metrics/issues/1093

Honestly, to me this doesn't seem as bad as ksm crashlooping, but would this cause other issues? Are these logs monitored? Would this fill up a volume?

---

Thanks guys for heads up, well spotted, testing like https://gigantic.slack.com/archives/C90BZMUJ1/p1587651052118200?thread_ts=1587390760.075200&cid=C90BZMUJ1 cannot spot that kind of issue, would be great to have this testing automated.

Found related issue / support request and suggestion to use 1.8.* for 1.15:
https://github.com/kubernetes/kube-state-metrics/issues/1093#issuecomment-596837244
which is aligned with the [compatibility matrix](https://github.com/kubernetes/kube-state-metrics#compatibility-matrix).

This probably affects all supported 1.15 based releases. We practically need separate ksm 1.15 compatibility branch if none of existing releases works. In that branch (if turned out to be need) we should still try to make all resources (other than binary) compatible with k8s 1.16 so that we have no problem with stuck helm releases during upgrade to k8s 1.16 based release.

Noting once more, 1.9.2 is not an option as well since it's crashing and paging https://github.com/giantswarm/giantswarm/issues/9506 and by the matrix 1.9.* binaries are not compatible with k8s 1.15

---

So can we just keep it as the same version we used for `9.0.0`?
https://github.com/giantswarm/releases/blob/master/release-notes/kvm/v9.0.0.md#kube-state-metrics-v180-gs-v060

---

I deployed a new release version with the old KSM on gorgoth. KSM logs have no errors, only one warning:
`W0424 09:05:19.624035       1 client_config.go:541] Neither --kubeconfig nor --master was specified.  Using the inClusterConfig.  This might not work.`

---

I've updated this release to include 1.8.0 / GS 0.6.0 kube-state-metrics

---

It solves for having binary compatible with k8s 1.15, I'm concerned that it could cause upgrade issues later - resources in the ksm 0.6.0 chart may be not compatible with k8s 1.16, and helm release could end up being stuck, only recovery being to delete and re-install it post upgrade.
Since ksm is more relevant to us than it us to customer, even that concern is not an issue/blocker (workaround with recreate is acceptable mitigation), as long as issue can be detected at least. Prerequisite for that is to have ksm specific AppFailed alert defined and page or at least notify. See more related alerts https://github.com/giantswarm/g8s-prometheus/blob/master/helm/g8s-prometheus/prometheus-rules/app.rules.yml That can be done as followup. Please consider adding a todo for this in team ludacris board.